src/Extensions/Subsites/IndexConfigurationExtension.php

```diff
@@ -16,8 +19,22 @@ public function updateIndexesForDocument(DocumentInterface $doc, array &$indexes
             return;
         }
 
-        // Which whether the data object has the SubsiteID
-        if (!$doc->getDataObject()->hasField('SubsiteID')) {
+        $dataObject = null;
+
+        try {
+            $dataObject = $doc->getDataObject();
+        } catch (Throwable $e) {
+            // if a data object has been deleted then an exception is thrown,
+            // but in the case of a non-versioned data object that has been deleted,
+            // we still need to remove it from the index
+            if (DataObject::has_extension($doc->getSourceClass(), Versioned::class)) {
+                throw $e;
+            }
+        }
+
+        // Check whether the data object has a Subsite ID
+        // @TODO: if a non versioned data object has been deleted, how do we know whether it had a subsite
```

Comment:
On further inspection, it looks like only Sitetree and child classes have the SubsiteID and it kinda looks like other data objects get indexed to all indexes.